src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -54,6 +54,23 @@
 @SuppressWarnings({"SqlDialectInspection", "SqlNoDataSourceInspection"})
 public class TestDualConnection {
 
+    @Test
+    public void shouldUseMainConnectionForInserts() throws SQLException {
+        final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
+        final Connection connection = DualConnection.builder(
+            connectionProvider,
+            permanentConsistency().build()
+        ).build();
+
+        connection.prepareStatement("insert into \"public\".\"fvv\" (\"cd\", \"iid\", \"did\", \"s\")\n" +
```

Comment:
Did that happens on production and results with failures from Replica ?
Asking cause I haven't see it yet.

Reply:
Yes. See the internal stand-up channel.

Reply:
`"ERROR: cannot execute INSERT in a read-only transaction"`

I see